### PR TITLE
[iOS] unwrap token for withdraw

### PR DESCRIPTION
**Pull request by thangpham26394**

### CODE REVIEW

File: EarnModule/EarnModule/ViewModel/UnstakeViewModel.swift [hunk 1]
```diff
@@ -52,11 +53,20 @@ class UnstakeViewModel: BaseViewModel {
...
+    var wrapInfo: WrapInfo?
     let apiService = EarnServices()
+    
+    var isLido: Bool {
+        return platform.name.lowercased() == "LIDO".lowercased()
+    }
+    
+    var isAnkr: Bool {
+        return platform.name.lowercased() == "ANKR".lowercased()
+    }
```
Comment (tungnguyen20) [inline EarnModule/EarnModule/ViewModel/UnstakeViewModel.swift:65]: I think should use `SpecifiedEarningPlatform`

Comment (thangpham26394) [reply EarnModule/EarnModule/ViewModel/UnstakeViewModel.swift:65]: this name is used from api response, it's a string and current model is struct parse directly from response so I think we can check by string for simplicity

File: EarnModule/EarnModule/ViewModel/UnstakeViewModel.swift [hunk 2]
```diff
@@ -52,11 +53,20 @@ class UnstakeViewModel: BaseViewModel {
...
     var buildTxRequestParams: JSONDictionary {
+        
         var earningType: String = platform.type
-        if toTokenSymbol.lowercased() == "MATIC".lowercased() {
+        if toTokenSymbol.lowercased() == "MATIC".lowercased() && ( isLido || isAnkr ) {
```
Comment (tungnguyen20) [inline EarnModule/EarnModule/ViewModel/UnstakeViewModel.swift:69]: Should not have space between ( ) and text

File: EarnModule/EarnModule/ViewModel/UnstakeViewModel.swift [hunk 3]
```diff
@@ -52,11 +53,20 @@ class UnstakeViewModel: BaseViewModel {
     var quoteTokenDetail: TokenDetailInfo?
     weak var delegate: UnstakeViewModelDelegate?
     var approveHash: String?
-    
+    var wrapInfo: WrapInfo?
     let apiService = EarnServices()
+    
+    var isLido: Bool {
+        return platform.name.lowercased() == "LIDO".lowercased()
```
Comment (tungnguyen20) [inline EarnModule/EarnModule/ViewModel/UnstakeViewModel.swift:60]: .uppercase = "LIDO"

File: EarnModule/EarnModule/ViewModel/UnstakeViewModel.swift [hunk 4]
```diff
@@ -245,12 +252,38 @@ class UnstakeViewModel: BaseViewModel {
...
+    func updateWrapInfo(isUseWrap: Bool) {
+        if isUseWrap && toTokenSymbol.first?.uppercased() == "W" {
+            toTokenSymbol.removeFirst()
+        } else {
+            toTokenSymbol = "W" + toTokenSymbol
+        }
+        
+        if let wrapInfo = wrapInfo {
+            toUnderlyingTokenAddress = wrapInfo.wrapAddress
+        }
```
Comment (tungnguyen20) [inline EarnModule/EarnModule/ViewModel/UnstakeViewModel.swift:264]: I think we should create two property: `wrappedSymbol` and `symbol`, then set respective value to `toTokenSymbol`. Put this logic here looks not clean.

File: EarnModule/EarnModule/ViewModel/UnstakeViewModel.swift [hunk 5]
```diff
@@ -245,12 +252,38 @@ class UnstakeViewModel: BaseViewModel {
...
+    func fetchData(isUseWrapTokenAddress: Bool = false, completion: @escaping () -> ()) {
+        var tokenAddress = toUnderlyingTokenAddress
+        if isUseWrapTokenAddress {
+            if let wrapInfo = wrapInfo {
+                tokenAddress = wrapInfo.wrapAddress
+            }
+        }
+        
+        
+        apiService.getStakingOptionDetail(platform: platform.name, earningType: platform.type, chainID: "\(chain.getChainId())", tokenAddress: tokenAddress) { result in
```
Comment (tungnguyen20) [inline EarnModule/EarnModule/ViewModel/UnstakeViewModel.swift:277]: Do we need `[weak self]` here

**Review by tungnguyen20 — APPROVED** [review]

**Pull request merged**